src/Components/Publishing/Display/Canvas/CanvasVideo.tsx

```diff
@@ -1,21 +1,37 @@
 import React from "react"
 import styled from "styled-components"
+import track from "../../../../Utils/track"
 import { pMedia } from "../../../Helpers"
 
 interface VideoProps {
+  campaign: any
   src: any
 }
 
+@track()
 class DisplayCanvasVideo extends React.Component<VideoProps, any> {
+  private video: HTMLVideoElement
+
   constructor(props) {
     super(props)
+    this.onPlayVideo = this.onPlayVideo.bind(this)
     this.state = { isPlaying: false }
   }
 
-  onPlayVideo = e => {
+  @track(() => ({
+    action: "Display Play Video",
+    campaign_name: this.props.campaign.name,
+  }))
+  onPlayVideo(e) {
     e.preventDefault()
-    e.target.paused ? e.target.play() : e.target.pause()
-    this.setState({ isPlaying: !e.target.paused })
+    if (this.video) {
+      if (this.video.paused) {
```

Comment:
Nitpick: It's often easier to read flat if/else blocks so this could be `if (this.video && this.video.paused) {} else {}`

Reply:
Checking for the video here is to guard against the ref not being present -- this.video doesn't seem to exist until after render, so the video controls break without.